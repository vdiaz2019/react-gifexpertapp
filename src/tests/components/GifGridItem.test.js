import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {
    const title = 'title';
    const url = 'url';
    let wrapper = shallow(<GifGridItem title = {title} url = {url} />);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title = {title} url = {url} />);
    })

    test('Debe de mostrar <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('Debe de tener un imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');   
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Div debe de tener un animate__fadeIn', () => {
        const img = wrapper.find('div');   
        const className = img.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
})