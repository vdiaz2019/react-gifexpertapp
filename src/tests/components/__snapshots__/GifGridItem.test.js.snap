// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="url"
  />
  <p>
    title
  </p>
</div>
`;
